emperor/support_files/js/sceneplotview3d.js

```diff
@@ -899,6 +973,180 @@ define([
+   * Note that we created a group of selectable objects in the constructor so
+   * we don't have to check for geometry types, etc.
+   *
+   * @param {Array} collection An array of objects to highlight
+   * @param {Integer} color A hexadecimal-encoded color. For shaders we only
+   * use the first bit to decide if the marker is rendered in white or rendered
+   * with the original color.
+   *
+   * @return {Array} selected objects (after checking for visibility and
+   * opacity).
+   *
+   * @private
+   */
+  ScenePlotView3D.prototype._highlightSelected = function(collection, color) {
+    var i = 0, j = 0, selected = [];
+
+    if (this.UIState.getProperty('view.usesPointCloud')) {
+      for (i = 0; i < collection.length; i++) {
+        // for shaders we only care about the first bit
+        var indices, emissiveColor = color & 1;
+
+        // if there's no selection then update all the points
+        if (collection[i].userData.selected === undefined) {
+          indices = _.range(collection[i].geometry.attributes.emissive.count);
+        }
+        else {
+          indices = collection[i].userData.selected;
+        }
+
+        for (j = 0; j < indices.length; j++) {
+          if (collection[i].geometry.attributes.visible.getX(indices[j]) &&
+              collection[i].geometry.attributes.opacity.getX(indices[j])) {
+            collection[i].geometry.attributes.emissive.setX(indices[j],
+                                                            emissiveColor);
+          }
+        }
+
+        collection[i].geometry.attributes.emissive.needsUpdate = true;
+        selected.push(collection[i]);
+      }
+    }
+    else {
+      for (i = 0; i < collection.length; i++) {
+        var material = collection[i].material;
+
+        if (material.visible && material.opacity && material.emissive) {
+          collection[i].material.emissive.set(color);
+          selected.push(collection[i]);
+        }
+      }
+    }
+
+    return selected;
+  };
+
+
+  /**
+   *
+   * Adds the mouse selection events to the current view
+   *
+   * @param {node} $container The container to add the events to.
+   * @private
+   */
+  ScenePlotView3D.prototype._addSelectionEvents = function($container) {
+    var scope = this;
+
+    // There's three stages to the mouse selection:
+    //  mousedown -> mousemove -> mouseup
+    //
+    // The mousdown event is ignored unless the user is holding Shift. Once
+    // selection has started the rotation controls are disabled. The mousemove
+    // event continues until the user releases the mouse. Once this happens
+    // rotation is re-enabled and the selection box disappears. Selected
+    // markers are highlighted by changing the light they emit.
+    //
+    $container.on('mousedown', function(event) {
+      // ignore the selection event if shift is not being held or if parallel
+      // plots are being visualized at the moment
+      if (!event.shiftKey || scope.parallelController.enabled) {
```

Comment:
it would be cool to allow this in parallel plots .... just a thought

Reply:
I looked at this and it wasn't much harder to add support for parallel plots so this now works:

![parallel-plot](https://user-images.githubusercontent.com/375307/84455934-9016f080-ac13-11ea-8395-bfff0eb967e1.gif)